src/main/java/it/gov/pagopa/pu/debtpositions/service/update/DebtPositionCancelInstallmentServiceImpl.java

```diff
@@ -29,23 +32,41 @@
 @Slf4j
 public class DebtPositionCancelInstallmentServiceImpl extends BaseDebtPositionOperationService implements DebtPositionCancelInstallmentService {
 
+  private final InstallmentDeletionService installmentDeletionService;
+  private final DebtPositionDeletionService debtPositionDeletionService;
+
   protected DebtPositionCancelInstallmentServiceImpl(AuthorizeOperatorOnDebtPositionTypeService authorizeOperatorOnDebtPositionTypeService,
                                                      DebtPositionService debtPositionService,
                                                      DebtPositionSyncService debtPositionSyncService,
                                                      DebtPositionProcessorService debtPositionProcessorService,
                                                      OrganizationService organizationService,
-                                                     DebtPositionHierarchyStatusAlignerService debtPositionHierarchyStatusAlignerService) {
+                                                     DebtPositionHierarchyStatusAlignerService debtPositionHierarchyStatusAlignerService,
+                                                     InstallmentDeletionService installmentDeletionService, DebtPositionDeletionService debtPositionDeletionService) {
     super(authorizeOperatorOnDebtPositionTypeService, debtPositionService, debtPositionSyncService, debtPositionProcessorService, organizationService, debtPositionHierarchyStatusAlignerService);
+    this.installmentDeletionService = installmentDeletionService;
+    this.debtPositionDeletionService = debtPositionDeletionService;
   }
 
   @Transactional
   @Override
   public WorkflowCreatedDTO cancelInstallment(DebtPositionDTO debtPositionDTO, List<InstallmentDTO> installments2operate, WfExecutionParameters wfExecutionParameters, String accessToken, String operatorExternalUserId) {
+    Set<Long> installmentIds = installments2operate.stream().map(InstallmentDTO::getInstallmentId).collect(Collectors.toSet());
+
     if (log.isDebugEnabled()) {
-      Set<Long> installmentIds = installments2operate.stream().map(InstallmentDTO::getInstallmentId).collect(Collectors.toSet());
       log.debug("Cancelling installments with ids {} for debt position with id {}", installmentIds, debtPositionDTO.getDebtPositionId());
     }
 
+    if (DebtPositionStatus.DRAFT.equals(debtPositionDTO.getStatus())) {
+      boolean allInstallmentsToBeDeleted = installmentIds.size() ==
+        debtPositionDTO.getPaymentOptions().stream()
+          .mapToLong(po -> po.getInstallments().size())
+          .sum();
+
+      if (allInstallmentsToBeDeleted) {
+        return debtPositionDeletionService.deleteDebtPosition(debtPositionDTO.getDebtPositionId(), accessToken, operatorExternalUserId);
+      }
+    }
```

Comment:
you have to do this on the `applyOperation` method, otherwise you will not check the operator permissions